core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -215,6 +215,13 @@ class TopicDeletionManager(controller: KafkaController,
       false
   }
 
+  def getPartitionsToBeDeleted(): Set[TopicAndPartition] = {
+    if(isDeleteTopicEnabled) {
+      partitionsToBeDeleted
+    } else
+      mutable.Set.empty[TopicAndPartition]
```

Comment:
Currently ControllerChannelManager class  calls this method like this "controller.deleteTopicManager.getPartitionsToBeDeleted". 
If we add Option here, then again we need to use getOrElse()/get() on this method. 
just want to avoid  long method calls.
 
I can change, if you still prefer option usage.